modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java

```diff
@@ -0,0 +1,251 @@
+         */
+        protected void addUnit(Unit<?> unit) {}
+
+        @Override
+        public void label(Unit<?> unit, String label) {
+            super.label(unit, label);
+            this.unitToName.put(unit, label);
+        }
+
+        public Appendable format(Unit<?> unit, Appendable appendable) throws IOException {
+            String name = this.nameFor(unit);
+            if (name != null) {
+                return appendable.append(name);
+            } else if (!(unit instanceof ProductUnit)) {
+                throw new IllegalArgumentException("Cannot format given Object as a Unit");
+            } else {
+                ProductUnit<?> productUnit = (ProductUnit) unit;
+                boolean start = true;
+
+                int i;
+                int pow;
+                int root;
+                for (i = 0; i < productUnit.getUnitCount(); ++i) {
+                    pow = productUnit.getUnitPow(i);
+                    if (!start) {
+                        appendable.append('*');
+                    }
+
+                    name = this.nameFor(productUnit.getUnit(i));
+                    root = productUnit.getUnitRoot(i);
+                    this.append(appendable, name, pow, root);
+                    start = false;
+                }
+
+                return appendable;
+            }
+        }
+
+        private void append(Appendable appendable, CharSequence symbol, int pow, int root)
+                throws IOException {
+            appendable.append(symbol);
+            if (pow != 1 || root != 1) {
+                appendable.append('^');
+                appendable.append(String.valueOf(pow));
+                if (root != 1) {
+                    appendable.append(':');
+                    appendable.append(String.valueOf(root));
+                }
+            }
+        }
+
+        protected String nameFor(Unit<?> unit) {
+            String label = this.unitToName.get(unit);
+            if (label != null) {
+                return label;
+            } else if (unit instanceof BaseUnit) {
+                return unit.getSymbol();
+            } else if (unit instanceof AlternateUnit) {
+                return unit.getSymbol();
+            } else if (!(unit instanceof TransformedUnit)) {
+                return null;
+            } else {
+                TransformedUnit<?> tfmUnit = (TransformedUnit) unit;
+                Unit<?> baseUnit = tfmUnit.getParentUnit();
+                UnitConverter cvtr = tfmUnit.getConverter();
+                StringBuilder result = new StringBuilder();
+                // this is the one line that needs to be replaced
+                String baseUnitName = format(baseUnit);
+                String prefix = this.prefixFor(cvtr);
+                if (baseUnitName.indexOf(183) < 0
+                        && baseUnitName.indexOf(42) < 0
+                        && baseUnitName.indexOf(47) < 0) {
+                    result.append(baseUnitName);
```

Comment:
Curiosity of mine... not required to approve the PR :) ... what are these numbers?

Reply:
I got it from a de-compiled version of the original library cause I had to change a single line of code to make parse/encode round trip actually work (they are not eager to fix it there, want use to upgrade to a different library that will be released in the future). Guess I can go and find the actual code, and see if I can use constants instead.

Reply:
No need for that... It was simply curiosity of mine :)